girder/molecules/server/calculation.py

```diff
@@ -296,15 +298,22 @@ def create_calc(self, params):
             paramType='body'))
 
     def _file_to_cjson(self, file, code='nwchem'):
+        readers = {
+            'nwchem': oc.NWChemJsonReader,
+            'psi4': oc.Psi4Reader
+        }
+
+        if code not in readers:
+            raise Exception('Unknown code %s' % code)
+        reader = readers[code]
+
         with File().open(file) as f:
             calc_data = f.read().decode()
-            if code == 'nwchem':
-                cjson = avogadro.convert_str(calc_data, 'json', 'cjson')
-                cjson = json.loads(cjson)
-            elif code == 'psi4':
-                cjson = cclib.convert_str(calc_data)
-            else:
-                raise Exception('Unknown code %s' % code)
+
+        with tempfile.TemporaryFile('w+') as tf:
```

Comment:
Couldn't we do this in memory ```BinaryIO```?

Reply:
Good point, I am now using the `SpooledTemporaryFile` class, that switches between `BytesIO` or `StringIO` to actual disk file when appropriate

Reply:
Good plan